trace/google/cloud/trace/trace_span.py

```diff
@@ -0,0 +1,153 @@
+                 For example, two spans with the same name may be
+                 distinguished using RPC_CLIENT and RPC_SERVER to identify
+                 queueing latency associated with the span.
+
+    :type parent_span_id: str
+    :param parent_span_id: ID of the parent span. Optional.
+
+    :type labels: dict
+    :param labels: Collection of labels associated with the span.
+                   Label keys must be less than 128 bytes.
+                   Label values must be less than 16 kilobytes.
+
+    :type start_time: str
+    :param start_time: (Optional) Start of the time interval (inclusive)
+                       during which the trace data was collected from the
+                       application.
+
+    :type end_time: str
+    :param end_time: (Optional) End of the time interval (inclusive) during
+                     which the trace data was collected from the application.
+
+    :type span_id: str
+    :param span_id: Identifier for the span, unique within a trace.
+    """
+
+    def __init__(
+            self,
+            name,
+            kind=Enum.SpanKind.SPAN_KIND_UNSPECIFIED,
+            parent_span_id=None,
+            labels=None,
+            start_time=None,
+            end_time=None,
+            span_id=None):
+        self.name = name
+        self.kind = kind
+        self.parent_span_id = parent_span_id
+        self.labels = labels
+        self.start_time = start_time
+        self.end_time = end_time
+
+        if span_id is None:
+            span_id = generate_span_id()
+
+        self.span_id = span_id
+
+        self.child_spans = []
+
+    def span(self, name='child_span'):
+        """Create a child span for the current span and append it to the child
+        spans list.
+
+        :type name: str
+        :param name: (Optional) The name of the child span.
+
+        :rtype: :class: `~google.cloud.trace.trace_span.TraceSpan`
+        :returns: A child TraceSpan to be added to the current span.
+        """
+        child_span = TraceSpan(name, parent_span_id=self.span_id)
+        self.child_spans.append(child_span)
+        return child_span
+
+    def set_start_time(self):
+        """Set the start time for a span."""
+        self.start_time = datetime.utcnow().isoformat() + 'Z'
+
+    def set_end_time(self):
+        """Set the end time for a span."""
+        self.end_time = datetime.utcnow().isoformat() + 'Z'
+
+    def __enter__(self):
+        self.set_start_time()
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.set_end_time()
+
+
+def generate_span_id():
+    """Return the random generated span ID for a span.
+
+    :rtype: int
+    :returns: Identifier for the span. Must be a 64-bit integer other
+                than 0 and unique within a trace. Converted to string.
+    """
+    span_id = str(random.getrandbits(64))
+    return int(span_id)
+
+
+def format_span_json(span):
```

Comment:
Yeah, but since we're using gRPC (and gRPC only), there's no need to convert to/from JSON - can we just use the protos directly?